manual/src/threelang/tfchain/tfchain.md

```diff
@@ -0,0 +1,237 @@
+    - `destination`: tfchain address in SS58 format
+- Swap balance to stellar account
+    ```md
+    !!tfchain.balance.swap
+        amount:100
+        destination:GCCVPYFOHY7ZB7557JKENAX62LUAPLMGIWNZJAFV2MITK6T32V37KEJU
+    ```
+    - `amount`: the amount of sent TFTs
+    - `destination`: stellar address
+- Get current balance for tfchain account
+    ```md
+    !!tfchain.balance.get
+        address:5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY   
+    ```
+    - `address`: tfchain address in SS58 format
+  
+## Twins
+- Get a twin info with different parameters
+    ```
+    !!tfchain.twins.get
+        id:29
+    ```
+    ```
+    !!tfchain.twins.get
+        pubkey:5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
+    ```
+    You can get the twin info by only providing one of the following:
+    - `id`: twin id
+    - `pubkey`: SS58 address for the twin
+
+- Create new twin on the chain for your account.
+    ```md
+    !!tfchain.twins.create
+        network:devnet
+        pubkey:5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
+    ```
+    - `network`: is the tfchain network, should be one of (mainnet, testnet, qanet, devnet)
+    - `pubkey`: SS58 address for the twin
+    
+
+## Nodes
+- Get nodes info from the chain
+    ```md
+    !!tfchain.nodes.get
+        farm_id:1
+    ```
+    This will return all the nodes for this farm.
+    ```
+    !!tfchain.nodes.get
+        node_id:11 
+    ```
+    This will only return the node with the same id.
+
+## Farms
+- Get farms info
+    ```
+    !!tfchain.farms.get
+        id:1
+    ```
+    ```
+    !!tfchain.farms.get
+        name:freefarm
+    ```
+    - `id`: is the farm id
+    - `name`: the full farm name "case-sensitive"
+- Create new farm on the chain
+    ```md
+    !!tfchain.farms.create
+        name:newfarm
+    ```
+    ```
+    !!tfchain.farms.create
+        name:newfarmwithips
+        public_ips: 185.206.122.152/16
+        gateways: 185.206.122.152
+    ```
+    It is optional to add ips to the farm
+    - `public_ips`: is a list of IP addresses in CIDR format xxx.xxx.xxx.xxx/xx. separated by comma.
+    - `gateways`: is a list of Gateways for the IP in ipv4 format. separated by comma.
+    Values on both public_ips/gateways lists are mapped by their indices. so for example gateways[0] is the gateway for the ip public_ips[0] and so on. 
+
+## Contracts
+- Get the contract info
+    ```
+    !!tfchain.contracts.get
+        id:6142
+    ```
+    Get specific contract by it's id.
+    ```
+    !!tfchain.contracts.get
+        node_id:11
+    ```
+    Get all contracts deployed on specific node.
+    ```
+    !!tfchain.contracts.get
+        node_id:11
+        hash:ff38eba6dbe9c5d2a1eb829bc48d77a6
+    ```
+    Get specific contract by it's hash and the node where it is deployed.
+
+- Create new Name contract
+    ```
+    !!tfchain.contracts.create_name
+        name:foo
+    ```
+
+- Create new Node contract
+    ```
+    !!tfchain.contracts.create_node
```

Comment:
I would rename to tfchain.contracts.create and provide the type of contract in a parameter type. For example:
```
!!tfchain.contracts.create
      node_id: 11
      type: node
      body: some body
      hash: some hash
      public_ips: 1
      solution_provider_id: 102

!!tfchain.contracts.create
      node_id: 11
      type: rent

!!tfchain.contracts.create
      type: name
      name: foo
```